gripper_press.go

```diff
@@ -96,25 +135,155 @@ func (g *myGripperPress) Grab(ctx context.Context, extra map[string]interface{})
 	if !force(extra) && !g.open {
 		return false, nil
 	}
+
+	// If the "Pin" field is set, only use that pin to control grab/open
+	if len(g.pins) != 0 {
+		err := g.setPins(ctx, g.pins, true, extra)
+		if err != nil {
+			return false, err
+		}
+		g.open = false
+		// Return early if no grab time is specified
+		if *g.conf.Seconds == 0 {
+			return false, nil
+		}
+		time.Sleep(time.Second * time.Duration(*g.conf.Seconds))
+		err = g.setPins(ctx, g.pins, false, extra)
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	}
+
+	// Set any wait pins to their active state
+	if len(g.conf.WaitPins) > 0 {
+		err := g.setPins(ctx, g.conf.WaitPins, true, extra)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	// Set open pins to their inactive state
+	err := g.setPins(ctx, g.conf.OpenPins, false, extra)
```

Comment:
interesting, i wasn't thinking of it this way.
in my mind, everything you need to grab is in grab pins